queue.h

```diff
@@ -0,0 +1,48 @@
+#ifndef CONCURRENT_QUEUE_H
```

Comment:
May I suggest keeping the `concurrentqueue.h` name, but placing the header/source in a folder named 'c-api'?

Reply:
ok, should it still defined both blocking and non-blocking in the same header ? if not then a third header that defines the export stuff should be created?

Reply:
I think it's ok to define just one header with both, named `concurrentqueue.h`